Sources/GenericModule/Module/Module.swift

```diff
@@ -39,8 +40,12 @@ open class Module<Presenter: ModulePresenter> where Presenter.View.ViewModel ==
     }
 
     public init(state: State, dependencies: Dependencies, output: Output? = nil) {
-        let viewModel = ViewModel(state: state)
         let presenter = Presenter(state: state, dependencies: dependencies)
+        guard let viewModelDelegate = presenter.makeViewModelDelegate() as? ViewModelDelegate else {
```

Comment:
I think this `as? ViewModelDelegate` can be removed by adding `Presenter.ViewModel.ViewModelDelegate == Presenter.ViewModelDelegate` condition